src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -233,6 +238,40 @@ private async Task RunLoop(CancellationToken cancellationToken)
                 }
             }
 
+            TimeSpan cancellationTime = TimeSpan.Parse(_options.Value.LogScrapingTimeout ?? "00:10:00");
```

Comment:
It feels like all this new code should probably be in another method.  It's pretty self contained, and could easily be tucked away in something like "AddImageNamesToRecordsAsync()" or something, then it's cancellation stuff would be easier to think about.  Trying to figure out which paths here result in which things happening in the containing function is a little mind boggling.

Reply:
Makes sense, extracted everything into a separate method